app/controllers/gladiators_controller.rb

```diff
@@ -0,0 +1,74 @@
+class GladiatorsController < ApplicationController
+  before_action :set_gladiator, only: [:show, :edit, :update, :destroy]
+  before_action :authenticate_admin!, except: [:show, :index]
+  # GET /gladiators
+  # GET /gladiators.json
+  def index
+    @gladiators = Gladiator.all
+  end
+
+  # GET /gladiators/1
+  # GET /gladiators/1.json
+  def show
+  end
+
+  # GET /gladiators/new
+  def new
+    @gladiator = Gladiator.new
+  end
+
+  # GET /gladiators/1/edit
+  def edit
+  end
+
+  # POST /gladiators
+  # POST /gladiators.json
+  def create
+    @gladiator = Gladiator.new(gladiator_params)
+    @competition.admin = current_admin
```

Comment:
Typo? We don't have an instance variable `@competition` here.
You can associate Gladiator' class children attributes like this:
```
Gladiator.new(gladiator_params, admin: current_admin)
```
knowing that Gladiator has an association with Admin.